src/celestia/CelestiaCoreApplication.h

```diff
@@ -0,0 +1,159 @@
+#ifndef __CELESTIA_CORE_APPLICATION__H__ 
+#define __CELESTIA_CORE_APPLICATION__H__
+
+//#include <cstring>
+#include "celestiacore.h"
+
+class CelestiaCoreApplication : public CelestiaCore {
+    bool autoMessages;
+    
+public:
+    
+    CelestiaCoreApplication();
+    
+    void setAutoMessagesOn() { autoMessages = true; }
+    void setAutoMessagesOff() { autoMessages = false; }
+    bool getAutoMessages();
+    void aflash(const char *n) {
+        if (getAutoMessages())  flash(_(n));
+    }
+    
+    template <int F> class CelestiaRenderFlag {
+        int flag;
+        CelestiaCoreApplication *core;
+        const char *onMsg;
+        const char *offMsg;
+    
+    public:
+        CelestiaRenderFlag(
+            CelestiaCoreApplication *c,
+            const char *on = 0, 
+            const char *off = 0) : 
+                core(c),
+                onMsg(on),
+                offMsg(off)
+        {}
+    
+        void flashOn() { if (onMsg) core->aflash(onMsg); }
+        void flashOff() { if (offMsg) core->aflash(offMsg); }
+    
+        void setOn() {
+            core->getRenderer()->setRenderFlags(core->getRenderer()->getRenderFlags() | F);
+            core->notifyWatchers(RenderFlagsChanged);
+        }
+    
+        void setOff() {
+            core->getRenderer()->setRenderFlags(core->getRenderer()->getRenderFlags() & ~F);
+            core->notifyWatchers(RenderFlagsChanged);
+        }
+    
+        void set(bool on) {
+            if (on) setOn();
+            else setOff();
+        }
+    
+        void toggle() {
+            core->getRenderer()->setRenderFlags(core->getRenderer()->getRenderFlags() ^ F);
+            core->notifyWatchers(RenderFlagsChanged);
+        }
+    
+        bool isSet() { return core->getRenderer()->getRenderFlags() & F; }
+    };
+    
+    // Text enter mode functions
+    
+    void setTextEnterModeOn() { setTextEnterMode(getTextEnterMode() | KbAutoComplete); }
+    
+    void setTextEnterModeOff() {
+        setTextEnterMode(KbNormal);
+        aflash("Cancel");
+    }
+    
+    // Selection functions
+    
+    void selectParentObject() { getSimulation()->setSelection(getSimulation()->getSelection().parent()); }
+    
+    void selectStar(int index) {
+        addToHistory();
+        getSimulation()->setSelection(getSimulation()->getUniverse()->getStarCatalog()->find(index));
+    }
+    
+    void centerSelection() {
+        addToHistory();
+        getSimulation()->centerSelection();
+    }
+    
+    // goto functions
+    
+    void gotoSelection(double gotoTime,
+                       const Eigen::Vector3f& up,
+                       ObserverFrame::CoordinateSystem upFrame);
+    
+    void gotoSelection(double gotoTime) {
+        gotoSelection(gotoTime, Eigen::Vector3f::UnitY(), ObserverFrame::ObserverLocal);
+    }
+    
+    void gotoSurface(double time = 5.0) {
+        aflash("Goto surface");
```

Comment:
Mark string for gettext (`N_()` afair)

Reply:
aflash() implementation do it already, however I can change it, if it's a bad idea.

Reply:
All strings I used are from legacy code, however I know little about used here translation method. I will take a look and check, how it works and what is needed to add. Thanks for help.